src/Microsoft.AspNetCore.Testing/ExceptionAssertions.cs

```diff
@@ -85,9 +83,10 @@ public static TException Throws<TException>(Func<object> testCode, string except
         /// <param name="exceptionMessage">The exception message to verify</param>
         /// <returns>The exception that was thrown, when successful</returns>
         /// <exception>Thrown when an exception was not thrown, or when an exception of the incorrect type is thrown</exception>
-        public static ArgumentException ThrowsArgument(Action testCode, string paramName, string exceptionMessage)
+        public static TException ThrowsArgument<TException>(Action testCode, string paramName, string exceptionMessage)
```

Comment:
Our typical pattern is `ExceptionAssert.ThrowsArgument(() => { }, "paramName");` so the compiler wouldn't be able to figure out the exception type. I'll have this fixed.